src/middlewared/middlewared/etc_files/local/smb4.conf

```diff
@@ -149,58 +166,64 @@
+                    'client ldap sasl wrapping': db['ad']['ldap_sasl_wrapping'].lower(),
+                    'template shell': '/bin/sh',
+                    'template homedir': f'{get_cifs_homedir()}/%D/%U',
+                    'ad allow dns update': 'Yes' if db['ad']['allow_dns_updates'] else 'No',
+                    'allow trusted domains': 'Yes' if db['ad']['allow_trusted_doms'] else 'No'
+                })
+                if not db['ad']['disable_freenas_cache']:
+                    pc.update({'winbind enum users': 'Yes'})
+                    pc.update({'winbind enum groups': 'Yes'})
                 if db['ad']['use_default_domain']:
                     pc.update({'winbind use default domain': 'Yes'})
-                pc.update({'winbind refresh tickets': 'Yes'})
                 if db['ad']['nss_info']:
                     pc.update({'winbind nss info': db['ad']['nss_info'].lower()})
-                if not db['ad']['allow_trusted_doms']:
-                    pc.update({'allow trusted domains': 'No'})
-                pc.update({'client ldap sasl wrapping': db['ad']['ldap_sasl_wrapping'].lower()})
-                pc.update({'template shell': '/bin/sh'})
-                pc.update({'template homedir': f'{get_cifs_homedir()}/%D/%U'})
 
             elif db['role'] == 'ldap_member':
-                pc.update({'server role': 'member server'})
-                pc.update({'security': 'user'})
-                if type(db['ldap']['ldap_hostname']) == list:
-                    ldap_pdb_backend = "ldapsam:\"%s\"" & " ".join([
-                        "%s://%s" % ("ldaps" if db['ldap']['ldap_ssl'] == 'ON' else "ldap", hostname)
-                        for hostname in db['ldap']['ldap_hostname'].split()
-                    ])
-                else:
-                    ldap_pdb_backend = f"ldapsam:{'ldaps' if db['ldap']['ldap_ssl'] == 'ON' else 'ldap'}://{db['ldap']['ldap_hostname']}"
-
+                pc.update({
+                    'server role': 'member server',
+                    'security': 'user',
+                    'ldap admin dn': db['ldap']['ldap_binddn'],
+                    'ldap suffix': db['ldap']['ldap_basedn'],
+                    'ldap user suffix': db['ldap']['ldap_usersuffix'],
+                    'ldap group suffix': db['ldap']['ldap_groupsuffix'],
+                    'ldap machine suffix': db['ldap']['ldap_machinesuffix'],
+                    'ldap replication sleep': '1000',
+                    'ldap passwd sync': 'Yes',
+                    'ldap ssl': 'off' if db['ldap']['ldap_ssl'] != 'START_TLS' else 'start tls',
+                    'ldapsam:trusted': 'Yes',
+                    'domain logons': 'Yes',
+                    'workgroup': db['cifs']['workgroup'].upper(),
+                })
+                ldap_pdb_backend = "ldapsam:\"%s\"" & " ".join([
+                    "%s://%s" % ("ldaps" if db['ldap']['ldap_ssl'] == 'ON' else "ldap", hostname)
```

Comment:
While you're here might as well convert it to .format() for f-string. `%` is deprecated in our code